src/main.ts

```diff
@@ -27,30 +27,77 @@ class ActiveSource
     }
 }
 
+interface Effect
+{
+    run(sample: number): number
+}
+
+class TremoloEffect
+{
+    rate: number
+    private counter: number = 0
+
+    constructor(rate: number) {
+        this.rate = rate
```

Comment:
private properties passed in constructor can be rewritten as:

constructor(private rate: number)
{
 // no this.rate = rate needed and no need to declare field in class body
}